DevEdu.API/DevEdu.Business/Services/LessonService.cs

```diff
@@ -35,49 +39,70 @@ public void AddCommentToLesson(int lessonId, CommentDto commentDto)
             _lessonRepository.AddCommentToLesson(lessonId, commentId);
         }
 
-        public int AddLesson(LessonDto lessonDto, List<int> topicIds)
+        public LessonDto AddLesson(UserDto userIdentity, LessonDto lessonDto, List<int> topicIds)
         {
+            _lessonValidationHelper.CheckUserAndTeacherAreSame(userIdentity, lessonDto.Teacher.Id);
+
             int lessonId = _lessonRepository.AddLesson(lessonDto);
 
             if (topicIds != null)
             {
-                topicIds.ForEach(topicId => _lessonRepository.AddTopicToLesson(lessonId, topicId));
+                foreach (int topicId in topicIds)
+                {
+                    _topicValidationHelper.CheckTopicExistence(topicId);
+                    _lessonRepository.AddTopicToLesson(lessonId, topicId);
+                }
             }
-
-            return lessonId;
+            return _lessonRepository.SelectLessonById(lessonId);
         }
 
         public void DeleteCommentFromLesson(int lessonId, int commentId) => _lessonRepository.DeleteCommentFromLesson(lessonId, commentId);
 
-        public void DeleteLesson(int id) => _lessonRepository.DeleteLesson(id);
-
-        public List<LessonDto> SelectAllLessonsByGroupId(int id) => _lessonRepository.SelectAllLessonsByGroupId(id);
-        
-        public List<LessonDto> SelectAllLessonsByTeacherId(int id) => _lessonRepository.SelectAllLessonsByTeacherId(id);
-        
-        public LessonDto SelectLessonById(int id) => _lessonRepository.SelectLessonById(id);
-        
-        public LessonDto SelectLessonWithCommentsById(int id)
+        public void DeleteLesson(UserDto userIdentity, int id)
         {
-            LessonDto result = _lessonRepository.SelectLessonById(id);
+            _lessonValidationHelper.CheckLessonExistence(id);
+            _lessonValidationHelper.CheckUserBelongsToLesson(userIdentity, id);
```

Comment:
here and in other places we wave two calls to _lessonRepository.SelectLessonById(lessonId);
the first one in CheckLessonExistence and the second in CheckUserBelongsToLesson
can we optimize it somehow?